### PR TITLE
xtensa-build-zephyr.py: small de-duplication

### PR DESCRIPTION
**Pull request by marc-hb**
~3 commits~
2 commits

### CODE REVIEW

**Comment by marc-hb** [comment]
sof-ci/jenkins is down for the week-end

**Review by lgirdwood — APPROVED** [review]

**Review by aborisovich — CHANGES_REQUESTED** [review]
Good refactor in general, only some remarks related to the comments wording and that exception catching.

File: scripts/xtensa-build-zephyr.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def parse_args():
 	if args.clone_mode and not args.z:
 		args.z = "main"		# set default name for -z if -c specified
 
-	if args.west_path:			# allow user override path to zephyr project
+	if args.west_path:			# allow re-use of existing zephyrproject/ anywhere
```
[the remark this answers is not in the document]
Comment (marc-hb) [reply scripts/xtensa-build-zephyr.py:188]: > Maybe its subjective but this wording is not intuitive for me.

Natural language is very rarely objective :-)

> I've been taught to not describe in the comments "what the code does" but rather "what is the purpose" of this code. "Re-use" in context of assigning to global variable just looks like you would be describing "how python works".

That's really not what I meant, I was referring neither to the code nor to python but to how the end-user can avoid cloning.

Unless you know the script well I think it's not obvious that --west-path is the option not to clone from scratch. How about this:

```suggestion
	if args.west_path: # lets the user provide an existing zephyrproject/ workspace anywhere
```

File: scripts/xtensa-build-zephyr.py [hunk 2]
```diff
@@ -266,13 +266,13 @@ def create_zephyr_sof_symlink():
 	try:
 		if not sof_symlink.is_symlink():
 			sof_symlink.symlink_to(SOF_TOP, target_is_directory=True)
-	except:
+	except BaseException as e:
```
[the remark this answers is not in the document]
Comment (marc-hb) [reply scripts/xtensa-build-zephyr.py:269]: > First of all, catching a BaseException is a work-around to silence pylint.

I didn't even run pylint, I just spotted this code first without pylint and then I googled the pylint warning. This sort of catch-all is bad in any language, so I knew pylint would have a warning for this and a better "commit message" than me. Every linter will flag this in any language.

> Correct me if I am wrong but I gave it at lot of thought to this line writing this.

I wasn't aware of these complexities. Silencing a warning makes sense only when there is comment that explains why + makes sure all your thoughts don't get lost!

> I found in this thread that you can just do except Exception: # pylint: disable=broad-except .

I'm not familiar enough with Windows to test this. I will drop this commit from this PR. Could you submit this when you get a chance?

PS: this script has 261 pylint warnings right now.
```
Your code has been rated at -1.26/10
```

File: scripts/xtensa-build-zephyr.py [hunk 3]
```diff
@@ -450,11 +442,17 @@ def main():
 	else:
 		print("Building platforms: {}".format(" ".join(args.platforms)))
 
-	# Two mutually exclusive working modes are supported
+	# argparse makes these two modes mutually exclusive.
```
[the remark this answers is not in the document]
Comment (marc-hb) [reply scripts/xtensa-build-zephyr.py:445]: Well of course the `argparse` authors have not written code to detect `--west-path` and `--clone-mode` and implement a mutual exclusion for free just for us, that seemed obvious to me but I can try to make that more explicit. My problem with the previous comment is that it did not gave any indication of why the two modes are mutually explicit, so it looked like a missing `else if` / bug.

How about this:
```suggestion
	#  we made these two modes mutually exclusive with argparse
```

File: scripts/xtensa-build-zephyr.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (aborisovich) [reply scripts/xtensa-build-zephyr.py]: Yeah I will check asap that exception.

> PS: this script has 261 pylint warnings right now.
Your code has been rated at -1.26/10

Well, it's not surprising given that this code was rewritten from bash script and not designed from scratch the "pythonic" way :/

File: scripts/xtensa-build-zephyr.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aborisovich) [reply scripts/xtensa-build-zephyr.py]: Yes this makes much more sense now! 👍

**Review by aborisovich — APPROVED** [review]

File: scripts/xtensa-build-zephyr.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (marc-hb) [reply scripts/xtensa-build-zephyr.py]: > Well, it's not surprising given that this code was rewritten from bash script and not designed from scratch the "pythonic" way :/

The vast majority of the pylint warnings are about indentation or mix of tabs and spaces. The second most frequent are about naming style.

[the remark this answers is not in the document]
Comment (aborisovich) [reply scripts/xtensa-build-zephyr.py]: Then looks like I need to improve my python styles, Installed pylint just now so expect improvements hoho!

[the remark this answers is not in the document]
Comment (marc-hb) [reply scripts/xtensa-build-zephyr.py]: > Then looks like I need to improve my python styles, Installed pylint just now so expect improvements hoho!

:-)

I think your coding style is fine, some of pylint choices are subjective, especially wrt. whitespace. I think there's a way to override some of them, maybe with a header. Please go easy on the whitespace, I have work in progress and others might have too! So I would prefer a header.

**Pull request merged**